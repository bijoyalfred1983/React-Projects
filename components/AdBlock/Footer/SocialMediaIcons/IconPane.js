/* @flow */

import React from 'react';
import type { Element } from 'react';
import styles from './styles.scss';

type Props = { btnContainer: Array<Object> };

const IconPane = ({ btnContainer }: Props): Element<'div'> => (
  <div className={styles.footerTop}>
    <ul className="">
      {btnContainer.map(navbutton => (
        <li key={navbutton.id}>
          <a href={navbutton.url}>
            <img src={navbutton.iconUrl} alt="" />
          </a>
        </li>
      ))}
    </ul>
  </div>
);

export default IconPane;
