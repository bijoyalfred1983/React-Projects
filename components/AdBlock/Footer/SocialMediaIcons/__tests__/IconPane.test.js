/* @flow */

import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import IconPane from '../IconPane';

describe('<IconPane />', () => {
  let PROPS;
  beforeEach(() => {
    PROPS = {
      btnContainer: [
        {
          id: '1',
          url: 'https://www.facebook.com/',
          iconUrl: '/images/socialmedia/icon-facebook.svg'
        }
      ]
    };
  });

  test('renders on Footer IconPane correctly', () => {
    const wrapper = mount(
        <IconPane {...PROPS} />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});


