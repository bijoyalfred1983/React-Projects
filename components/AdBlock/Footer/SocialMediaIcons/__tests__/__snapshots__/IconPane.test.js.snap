// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconPane /> renders on Footer IconPane correctly 1`] = `
<IconPane
  btnContainer={
    Array [
      Object {
        "iconUrl": "/images/socialmedia/icon-facebook.svg",
        "id": "1",
        "url": "https://www.facebook.com/",
      },
    ]
  }
>
  <div
    className="footerTop"
  >
    <ul
      className=""
    >
      <li>
        <a
          href="https://www.facebook.com/"
        >
          <img
            alt=""
            src="/images/socialmedia/icon-facebook.svg"
          />
        </a>
      </li>
    </ul>
  </div>
</IconPane>
`;
